### DISCUSSION TITLE
Remove an element if some condition is met

### DISCUSSION THREAD

**Discussion by L30Bola** [mikefarah/yq#2154] [Q&A] [answered]
Hi!!

I'd like some help on a yq command. Some context before:

I have a bunch of `spec.yaml` file that are structured like this:

```yaml
namespace: test-namespace
releaseName: test-release
paused: false
environment: production
config:
  region: us-east-1
```

I wanted to first add a `.config.image` field on the `production` specs, which I could successfully with this:

```bash
for yaml in directory/**/spec.yaml; do
    yq -i '(select(.environment == "production") | .config.image) = "alpine:latest"' $yaml
done
```

With the above for-loop, only the `production` specs were updated. Now, if I want to delete the added field, I tried this:

```bash
for yaml in directory/**/spec.yaml; do
    yq -i '(select(.environment == "production") | del(.config.image))' $yaml
done
```

And also this:

```bash
for yaml in directory/**/spec.yaml; do
    yq -i '(select(.environment == "production")) | del(.config.image)' $yaml
done
```

The problem is that, even though on `production` `spec.yaml` the `.config.image` field were deleted, `spec.yaml` that have `environment != production` had it's entire content deleted. Is there a way to delete the field on `production` `spec.yaml` while ignoring non-`production`?

The first part to add a new value, I got from this other discussion: https://github.com/mikefarah/yq/discussions/617

EDIT:
Forgot to add the `yq` version:

```bash
$ yq --version
yq (https://github.com/mikefarah/yq/) version v4.44.3
```

**Comment by mikefarah** [comment, marked as answer]
Sure, the issue is that you are first filtering to only include files with environment = production, and then deleting nodes in those matches. 

Instead do:

```
yq 'del(select(.environment == "production") | .config.image)' examples/data1.yaml
```

The filtering is _inside_ the delete, so it will only delete those matching nodes, or leave the data alone.

**Reply by L30Bola** [reply]
Oh, very nice. It worked flawlessly!
